vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1247,6 +1248,21 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private int getSiblWithReqPrimType(Node child) throws RepositoryException {
+
+        int count=0;
+        Node parent = child.getParent();
+        NodeIterator iter = parent.getNodes();
+        while (iter.hasNext()) {
+            Node sibling = iter.nextNode();
+            if(child.getPrimaryNodeType().toString()
```

Comment:
what about extended primary node types (https://docs.adobe.com/content/docs/en/spec/jsr170/javadocs/jcr-2.0/javax/jcr/nodetype/NodeType.html#isNodeType(java.lang.String))?

Reply:
actually, this more complicated, if same name siblings are allowed and stupidly complex CNDs....but let's ignore this for now :-)